pkg/cmd/dockerregistry/dockerregistry.go

```diff
@@ -45,8 +47,41 @@ import (
 	registryconfig "github.com/openshift/origin/pkg/dockerregistry/server/configuration"
 )
 
+var prune = flag.Bool("prune", false, "Prune blobs from the storage. DANGEROUS! Do it only in read only mode")
```

Comment:
Also if we could get blob's modification time, it would be a relatively safe operation even without switching to read-only mode. And it is relatively unsafe operation on an eventually consistent storage (like S3) even in read-only mode.